File: templates/problems/18_bit_manipulation/07_reverse_integer/reverse_integer_test.go
package reverse_integer

import "testing"

func TestReverse(t *testing.T) {
	tests := []struct {
		name     string
		x        int
		expected int
	}{
		{"positive", 123, 321},
		{"negative", -123, -321},
		{"trailing zero", 120, 21},
		{"zero", 0, 0},
		{"single digit", 5, 5},
		{"overflow positive", 1534236469, 0},
		{"overflow negative", -2147483648, 0},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := reverse(tt.x)
			if got != tt.expected {
				t.Errorf("reverse(%d) = %d, want %d", tt.x, got, tt.expected)
			}
		})
	}
}

File: templates/problems/18_bit_manipulation/07_reverse_integer/typescript/reverseInteger.test.ts
import { describe, it, expect } from 'vitest';
import { reverse } from './reverseInteger';

describe('ReverseInteger', () => {
    it('positive', () => {
        expect(reverse(123)).toBe(321);
    });

    it('negative', () => {
        expect(reverse(-123)).toBe(-321);
    });

    it('trailing zero', () => {
        expect(reverse(120)).toBe(21);
    });

    it('zero', () => {
        expect(reverse(0)).toBe(0);
    });

    it('single digit', () => {
        expect(reverse(5)).toBe(5);
    });

    it('overflow positive', () => {
        expect(reverse(1534236469)).toBe(0);
    });

    it('overflow negative', () => {
        expect(reverse(-2147483648)).toBe(0);
    });
});
